### ISSUE TITLE
sw.js on a different port in webpack

### ISSUE THREAD

**Issue by dawidvdh** [NekR/offline-plugin#76]
Hi, i am currently using a very similar webpack config to [This Demo Project](https://github.com/erikras/react-redux-universal-hot-example) and i currently need the service worker to run in dev mode.

the problem i'm experiencing is that the app is running on http://localhost:4000/ but the sw.js output is going to http://localhost:4001/dist/sw.js so when installing on the client it looks for the sw.js in http://localhost:4000/sw.js is there anyway around this?

My current Offline config:

```javascript
new OfflinePlugin({
      // publicPath: '../static/',
      relativePaths: false,
      ServiceWorker: {
        entry: './src/sw-handler.js'
      },
      AppCache: {
        directory: '../'
      }
    }),
```

**Comment by NekR** [comment]
I am not 100% sure I get what you are doing, but basically SW file cannot live on a separate domain. This is security limitation of SW. So, if your website lives on `example.com` then SW should be exactly on that domain, not `www.example.com`, not `cdn.example.com` and not `example.com:8181`.

This doesn't mean that you cannot place your files on CDN and then cache then in SW. **You can**. But again, SW itself should be on your domain and CDN assets should be served with CORS headers, otherwise SW cannot know if request was successful or not, so assets won't be cached.

Feel free to ask other questions if this isn't the answer to your issue. Thanks.

**Comment by dawidvdh** [comment]
Thanks for the quick response @NekR, I do understand that, I am just looking for a way to get this to work with that project...  is there a way to set the output manually? a separate config so that it isn't relative to webpack's config output.path?

**Comment by NekR** [comment]
@dawidvdh okay. I really have no idea what that project does and how it works. Also `offline-plugin` doesn't work with hot reloading. It's nearly impossible to get it to work there without bugs.

I can try help you of course here, but I need to know what that project does because it looks like you have some limitations and cannot change config as you want. Am I wrong here?

**Comment by NekR** [comment]
Hi @dawidvdh. Does it help you? If not, let's figure it out. Feel free to ask any questions.

**Comment by dawidvdh** [comment]
Hi @NekR sorry, i've been trying to come up with a solution, the best i could come up with was using a proxy, so when i request the service worker it pulls it from the webpack dev server running on the other port, are you able to get it working with that demo project i linked?

**Comment by NekR** [comment]
I am not sure if I wan't setting up that Demo because it looks too big. I'll think about it.

In a meantime, can you explain why the app lives on `http://localhost:4000/ ` but webpack's build goes to `http://localhost:4001/dist/`? I mean, why 2 servers at all? Is `http://localhost:4001/` mean to be a CDN/static-resources type domain on production? Why cannot you just move `sw.js` to `http://localhost:4000/ `?

If you can move `sw.js` to **4000** server then solution from this issue probably can help you too: https://github.com/NekR/offline-plugin/issues/65#issuecomment-223964545

Basically you will need config like this:

```js
new OfflinePlugin({
  publicPath: '/',
  relativePaths: false,
  rewrites: function(asset) {
    // This is code from default rewrites() function, feel free to remove it
    // if not needed
    asset = asset.replace(/^([\s\S]*?)index.htm(l?)$/, function(match, dir) {
      return dir || '/';
    });

    return '//localhost:4001/' + asset.replace(/^\//, '');
  }
})
```

I know this is kind of a mess, but this is the only solution at a moment :-(

**Comment by NekR** [comment]
Any updates? Should I close this?
